sycl/source/device_selector.cpp

```diff
@@ -157,5 +161,167 @@ int host_selector::operator()(const device &dev) const {
   return Score;
 }
 
+namespace ONEAPI {
+namespace detail {
+struct filter {
+  backend Backend = backend::host;
+  RT::PiDeviceType DeviceType = PI_DEVICE_TYPE_ALL;
```

Comment:
Any reason why you want to use this type instead of sycl::info::device_type?

Reply:
Efficiency.  I didn't want to have to repeatedly call "dev.get_info<info::device::device_type>()".

Reply:
As far as I can see, both RT::PiDeviceType and info::device_type are just enums, I believe they even use the same number to represent device types. Could you please elaborate more on what calls would you additionally need to do if you changed the type to info::device_type?

Reply:
There's no easy way to get info::device_type from a device other than the info query.  I can just get the RT::PiDeviceType from the device impl.

Reply:
Ok, I see, we could change the type of device_impl::MType as well. But, probably it should not be done in scope of this patch.